frontend/simple/js/state.js

```diff
@@ -48,15 +56,22 @@ const mutations = {
     // we've successfully received it back, so remove it from expectation pending
     const index = state.pending.indexOf(contractId)
     state.pending.includes(contractId) && state.pending.splice(index, 1)
+    // calling this will make pubsub subscribe for events on `contractId`!
+    sbl('vuex/v1/events/emit', 'contractsModified', {add: contractId})
   },
   setRecentHash (state, {contractId, hash}) {
     state.contracts[contractId] && Vue.set(state.contracts[contractId], 'recentHash', hash)
   },
   removeContract (state, contractId) {
     store.unregisterModule(contractId)
     Vue.delete(state.contracts, contractId)
+    // calling this will make pubsub unsubscribe for events on `contractId`!
+    sbl('vuex/v1/events/emit', 'contractsModified', {remove: contractId})
   },
   setContracts (state, contracts) {
+    for (let contractId of Object.keys(state.contracts)) {
+      mutations.removeContract(state, contractId)
```

Comment:
Because `commit` isn't passed to mutations, and yes, we could call `store.commit`, but why do that when we're already in the middle of the `setContracts` commit? It will have no benefit to do that AFAICT, and it would only clutter the vuex devtools log.